File: Golang/GrayCode/GrayCode_test.go
package GrayCode

import (
	"testing"
	"reflect"
)

func TestGrayCode(t *testing.T) {

	if reflect.DeepEqual(grayCode(0), []int{0}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(grayCode(2), []int{0, 1, 3, 2}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(grayCode(3), []int{0, 1, 3, 2, 6, 7, 5, 4}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/GrayCodeTest.py
from unittest import TestCase

from GrayCode import GrayCode


class TestGrayCode(TestCase):
    def test_grayCode(self):
        gc = GrayCode()

        self.assertEqual(gc.grayCode(0), [0])

        self.assertEqual(gc.grayCode(2), [0, 1, 3, 2])

        self.assertEqual(gc.grayCode(3), [0, 1, 3, 2, 6, 7, 5, 4])
